totalRP3/Core/Popup.lua

```diff
@@ -766,8 +766,25 @@ function TRP3_API.popup.showCompanionBrowser(onSelectCallback, onCancelCallback,
 	else
 		TRP3_CompanionBrowserTitle:SetText(loc.REG_COMPANION_BROWSER_MOUNT);
 	end
+
 	ui_CompanionBrowserContent.onSelectCallback = onSelectCallback;
 	ui_CompanionBrowserContent.onCancelCallback = onCancelCallback;
+
+	-- Do not steal input if we're in combat.
+	if not InCombatLockdown() then
+		local frame = TRP3_CompanionBrowser;
+		frame:SetScript("OnKeyDown", function(_, key)
+			if key == "ESCAPE" then
+				PlaySound(SOUNDKIT.IG_MAINMENU_OPTION_CHECKBOX_ON);
+				frame:SetPropagateKeyboardInput(false);
+				-- Hiding frames & Cancel callback is handled within.
+				onCompanionClose();
+			else
+				frame:SetPropagateKeyboardInput(true);
+			end
+		end);
```

Comment:
The InCombatLockdown check needs to be inside the SetScript callback, not outside.

Reply:
Should be resolved in latest commit, thank you.